### ISSUE TITLE
Redirect problem

### ISSUE THREAD

**Issue by soulgalore** [sitespeedio/sitespeed.io#2281]
`docker run --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io https://www.1177.se -n 1
Google Chrome 72.0.3626.81 
Mozilla Firefox 65.0
[2019-01-31 21:02:12] INFO: Versions OS: linux 4.9.125-linuxkit nodejs: v10.13.0 sitespeed.io: 8.0.6 browsertime: 4.1.0 coach: 3.1.2
[2019-01-31 21:02:12] INFO: Running tests using Chrome - 1 iteration(s)
[2019-01-31 21:02:13] INFO: Testing url https://www.1177.se iteration 1
[2019-01-31 21:02:27] INFO: https://www.1177.se/Stockholm/?ar=True 80 requests, 3016.97 kb, backEndTime: 421ms, firstPaint: 677ms, firstVisualChange: 800ms, DOMContentLoaded: 1.09s, Load: 1.37s, speedIndex: 988, visualComplete85: 1s, lastVisualChange: 1.27s, rumSpeedIndex: 737
[2019-01-31 21:02:28] ERROR: Error: ENOENT: no such file or directory, scandir '/sitespeed.io/sitespeed-result/www.1177.se/2019-01-31-21-02-12/pages/www.1177.se/data/video/images/1'`

**Comment by soulgalore** [comment]
I think this is what's happening: If we use a script/--multi we always use the URL from the browser, but not if we just supply one URL (to keep other plugins work https://github.com/sitespeedio/sitespeed.io/issues/2258#issuecomment-456424034).

**Comment by allangood** [comment]
Hi @soulgalore this problem happens to me for any website that redirects to a subfolder, like https://www.walmart.ca that redirects to https://www.walmart.ca/en or https://www.walmart.ca/fr. This is a big issue in Canada where almost all websites offer English and French content based on the browser language. :)

My run:
(On Ubuntu Xenial 16.04)
```
docker run --shm-size=1g --rm -v /opt/sitespeed:/sitespeed.io sitespeedio/sitespeed.io https://www.walmart.ca
Google Chrome 72.0.3626.81
Mozilla Firefox 65.0
[2019-02-07 18:21:07] INFO: Versions OS: linux 4.4.0-140-generic nodejs: v10.13.0 sitespeed.io: 8.2.3 browsertime: 4.2.5 coach: 3.2.0
[2019-02-07 18:21:07] INFO: Running tests using Chrome - 3 iteration(s)
[2019-02-07 18:21:08] INFO: Testing url https://www.walmart.ca iteration 1
[2019-02-07 18:21:39] INFO: https://www.walmart.ca/en BackEndTime: 270 DomInteractiveTime: 1403 DomContentLoadedTime: 1403 FirstPaint: 517 PageLoadTime: 3045
[2019-02-07 18:21:39] INFO: VisualMetrics FirstVisualChange: 1300 SpeedIndex: 4858 PerceptualSpeedIndex: 3346 LastVisualChange: 8200
[2019-02-07 18:21:41] INFO: Testing url https://www.walmart.ca iteration 2
[2019-02-07 18:22:15] INFO: https://www.walmart.ca/en BackEndTime: 166 DomInteractiveTime: 4398 DomContentLoadedTime: 4398 FirstPaint: 764 PageLoadTime: 8285
[2019-02-07 18:22:15] INFO: VisualMetrics FirstVisualChange: 1633 SpeedIndex: 7907 PerceptualSpeedIndex: 4683 LastVisualChange: 11900
[2019-02-07 18:22:16] INFO: Testing url https://www.walmart.ca iteration 3
[2019-02-07 18:22:46] INFO: https://www.walmart.ca/en BackEndTime: 85 DomInteractiveTime: 1843 DomContentLoadedTime: 1843 FirstPaint: 961 PageLoadTime: 5097
[2019-02-07 18:22:46] INFO: VisualMetrics FirstVisualChange: 1566 SpeedIndex: 5236 PerceptualSpeedIndex: 3673 LastVisualChange: 8666
[2019-02-07 18:22:46] INFO: https://www.walmart.ca/en 185 requests, 1530.43 kb, backEndTime: 174ms (±43.72ms), firstPaint: 747ms (±104.87ms), firstVisualChange: 1.50s (±83.03ms), DOMContentLoaded: 2.55s (±762.35ms), Load: 5.48s (±1.24s), speedIndex: 6000 (±783.48), visualComplete85: 9.45s (±953.57ms), lastVisualChange: 9.59s (±949.97ms), rumSpeedIndex: 748 (±104.14) (3 runs)
[2019-02-07 18:22:51] ERROR: Error: ENOENT: no such file or directory, scandir '/sitespeed.io/sitespeed-result/www.walmart.ca/2019-02-07-18-21-07/pages/www.walmart.ca/data/video/images/3'
```
Folder structure created:
```
find www.walmart.ca/2019-02-07-18-21-07/ -type d
www.walmart.ca/2019-02-07-18-21-07/
www.walmart.ca/2019-02-07-18-21-07/logs
www.walmart.ca/2019-02-07-18-21-07/pages
www.walmart.ca/2019-02-07-18-21-07/pages/www.walmart.ca
www.walmart.ca/2019-02-07-18-21-07/pages/www.walmart.ca/en
www.walmart.ca/2019-02-07-18-21-07/pages/www.walmart.ca/en/data
www.walmart.ca/2019-02-07-18-21-07/pages/www.walmart.ca/en/data/video
www.walmart.ca/2019-02-07-18-21-07/pages/www.walmart.ca/en/data/video/images
www.walmart.ca/2019-02-07-18-21-07/pages/www.walmart.ca/en/data/video/images/1
www.walmart.ca/2019-02-07-18-21-07/pages/www.walmart.ca/en/data/video/images/2
www.walmart.ca/2019-02-07-18-21-07/pages/www.walmart.ca/en/data/video/images/3
www.walmart.ca/2019-02-07-18-21-07/pages/www.walmart.ca/en/data/screenshots
www.walmart.ca/2019-02-07-18-21-07/pages/www.walmart.ca/data
```
The problem happens because sitespeed is creating the video under "/en" sub directory instead on parent folder.

Basically, it's not possible to run version > 8.0.0 in any site that redirects to a subfolder. For me, this is particular important, because I need to measure the time for this first redirect.

Hope this helps to find the bug!

Regards.

**Comment by soulgalore** [comment]
Hi @allangood thanks for adding the extra info! It's annoying. It actually worked in the first release (we used redirected URL as default then by default) but then it was problems with other plugins (WebPageTest/GPSI and Lighthouse see https://github.com/sitespeedio/sitespeed.io/issues/2258), so I've changed the default behaviour (and it bite back now). 

I've looked into change the default behaviour for the plugins and both Lighthouse and GPSI reported back the redirected URL but not WebPageTest (I think it did per run but not in the overall summary).

I think in the long run we need to change the behaviour or Lighthouse/GPSI and WebPageTest. Any change you have time to help me with that? I can guide you. Else it's something I can look at next week.

You are correct, the `--multi` switch uses the redirected URL and it works.
